### DISCUSSION TITLE
When I query a view on a collection filters do not work

### DISCUSSION THREAD

**Discussion by joshpetit** [Automattic/mongoose#12658] [Q&A] [answered]
Originally asked: https://stackoverflow.com/questions/74236990/how-to-filter-query-on-a-view-of-a-collection-in-mongoose

I have a model that's a view of another collection that I create like such:
```ts
  NewSchema.createCollection({
    viewOn: originalModel.collection.collectionName,
    pipeline: [
      {
        '$project': {
          uuid: 1,
          name: 1,
          description: 1,
          image_url: 1,
          price: 1,
          avg_rating: 1
      }
      },
    ],
  });
```
But when we do queries on the new schema, like this:
```ts
const res = await NewSchema.find({name: {$regex: keywords, $options: 'i' }}).sort({ 'price': -1 });
// or
const res = await NewSchema.find({uuid: "Something Random"});
```
The results contain everything in the original collection (the projection works, but the filter does not). When we do the query on the base collection, the filter works. Is it possible to filter a query with mongoose on a model that's a view of another schema?

**Comment by vkarpov15** [comment, marked as answer]
Try `mongoose.set('strictQuery', false)`?

**Reply by joshpetit** [reply]
Oh wow that seemed to have worked! What exactly is the reason why this option needs to be set? I'm assuming it's because maybe the fields don't exist in the view yet or something, new to mongodb so just curious 😅. Thank you!

**Reply by vkarpov15** [reply]
You're trying to query by a field that isn't defined in your schema. Admittedly Mongoose filtering out fields in the query that aren't in the schema is very confusing behavior, we're changing that in 7.0 with #11861 . When working with views, you typically would create a new model whose schema matches the fields in the view.
